### DISCUSSION TITLE
How to customize Button padding?

### DISCUSSION THREAD

**Discussion by azelenkevich** [ant-design/ant-design#39537] [Q&A]
I'm using AntD 5.0.2 and need to override Button component padding in ConfigProvider theme config. How can i do this? Overriding just "padding" property doesn't work

**Comment by Sparsh-Malhotra** [comment]
@azelenkevich Did you get it ?

**Comment by monolithed** [comment]
https://github.com/ant-design/ant-design/issues/41024

**Comment by orlandovallejos** [comment]
Any news on this?
This is supposed to be one of the easiest thing...

I want to update the padding of a button, how can I do it using the theme?
`controlPaddingHorizontal` property doesnt work

**Reply by monolithed** [reply]
There's nothing you can do, only override it  (

```css
.ant-btn {
    padding: 25px !important;
}
```

**Reply by orlandovallejos** [reply]
oh wow!
I thought there was a property or a way to easily do it in the theme in the latest version :( 

Anyway, thank you!

**Comment by Sanan4li** [comment]
Still waiting for a proper way to do it. I thought we don't need to use `!important` anymore when customizing. Anyways, our organization wants us to use ant-design over tailwind. Let' s do it in dirty way.

**Comment by vitalijalbu** [comment]
us `theme.json`

**Comment by JoeyAziz** [comment]
I wanted to apply custom padding to my buttons as well, so I had to dig deep into the code to find what controls the button padding

```
 <ConfigProvider
      theme={{
        components: {
          Button: {
            paddingInline: 100, // x-axis
            controlHeight: 100, // y-axis
          },
        },
      }}
    >
```

same applies for sizes 

![image](https://github.com/user-attachments/assets/e64ad407-98cd-4b0d-9d37-6664c5d1893d)
